build.go

```diff
@@ -267,10 +267,11 @@ func untar(dest string, r io.Reader) error {
 			if err != nil {
 				return err
 			}
-			defer f.Close()
 
 			// copy over contents
-			if _, err := io.Copy(f, tr); err != nil {
+			_, err = io.Copy(f, tr)
+			f.Close()
```

Comment:
can you add a comment as to why this is moved so its written in history and someone wont change it accidentally, thanks!

Reply:
Done.